### PR TITLE
[M68k] Emit RTE for interrupt handler.

### LINKED ISSUE #64833: [M68k] RTE instruction missing
The `rte` instruction is missing and is not emitted in `__attribute__((interrupt))` functions.

### PR DESCRIPTION
**Pull request by 0x59616e**
Fixes #64833

### CODE REVIEW

**Comment by glaubitz** [comment]
CC @thankfulmachine You may want to test this patch whether it addresses the original issue.

**Comment by glaubitz** [comment]
@mshockwave Any comment? I will test the changes later.

**Review by mshockwave — APPROVED** [review]
LGTM thanks for fixing this!

File: llvm/lib/Target/M68k/M68kExpandPseudo.cpp [hunk 1]
```diff
@@ -252,12 +252,11 @@ bool M68kExpandPseudo::ExpandMI(MachineBasicBlock &MBB,
...
-    // Adjust stack to erase error code
-    int64_t StackAdj = MBBI->getOperand(0).getImm();
-    MachineInstrBuilder MIB;
-
-    if (StackAdj == 0) {
-      MIB = BuildMI(MBB, MBBI, DL, TII->get(M68k::RTS));
+    if (MBB.getParent()->getFunction().getCallingConv() ==
+        CallingConv::M68k_INTR) {
+      BuildMI(MBB, MBBI, DL, TII->get(M68k::RTE));
+    } else if (int64_t StackAdj = MBBI->getOperand(0).getImm(); StackAdj == 0) {
```
Comment (RKSimon) [inline llvm/lib/Target/M68k/M68kExpandPseudo.cpp:258]: Why not just `} else if (MBBI->getOperand(0).getImm() == 0) {` ?

**Pull request merged**